plugin/mysql_extend/init.go

```diff
@@ -0,0 +1,185 @@
+				if !c.NextArg() {
+					return c.ArgErr()
+				}
+				userMaxIdleConns, err := strconv.Atoi(c.Val())
+				if err != nil || userMaxIdleConns <= zero {
+					m.maxIdleConns = defaultMaxIdleConns
+				} else {
+					m.maxIdleConns = userMaxIdleConns
+				}
+			case "db_max_open_conns":
+				if !c.NextArg() {
+					return c.ArgErr()
+				}
+				userMaxOpenConns, err := strconv.Atoi(c.Val())
+				if err != nil || userMaxOpenConns <= zero {
+					m.maxOpenConns = defaultMaxOpenConns
+				} else {
+					m.maxOpenConns = userMaxOpenConns
+				}
+			case "db_conn_max_idle_time":
+				if !c.NextArg() {
+					return c.ArgErr()
+				}
+				userConnMaxIdleTime, err := time.ParseDuration(c.Val())
+				if err != nil || userConnMaxIdleTime <= zeroTime {
+					m.connMaxIdleTime = defaultConnMaxIdleTime
+				} else {
+					m.connMaxIdleTime = userConnMaxIdleTime
+				}
+			case "db_conn_max_life_time":
+				if !c.NextArg() {
+					return c.ArgErr()
+				}
+				userConnMaxLifeTime, err := time.ParseDuration(c.Val())
+				if err != nil || userConnMaxLifeTime <= zeroTime {
+					m.connMaxLifetime = defaultConnMaxLifeTime
+				} else {
+					m.connMaxLifetime = userConnMaxLifeTime
+
+				}
+			case "fail_heartbeat_time":
+				if !c.NextArg() {
+					return c.ArgErr()
+				}
+				userFailHeartBeatTime, err := time.ParseDuration(c.Val())
+				if err != nil || userFailHeartBeatTime <= zeroTime {
+					m.failHeartbeatTime = defaultFailHeartBeatTime
+				} else {
+					m.failHeartbeatTime = userFailHeartBeatTime
+				}
+			case "success_heartbeat_time":
+				if !c.NextArg() {
+					return c.ArgErr()
+				}
+				userSuccessHeartBeatTime, err := time.ParseDuration(c.Val())
+				if err != nil || userSuccessHeartBeatTime <= zeroTime {
+					m.successHeartbeatTime = defaultSuccessHeartBeatTime
+				} else {
+					m.successHeartbeatTime = userSuccessHeartBeatTime
+				}
+
+			case "query_zone_sql":
+				if !c.NextArg() {
+					return c.ArgErr()
+				}
+				m.queryZoneSQL = c.Val()
+			case "query_record_sql":
+				if !c.NextArg() {
+					return c.ArgErr()
+				}
+				m.queryRecordSQL = c.Val()
+			default:
+				return c.Errf("unknown property '%s'", c.Val())
+			}
+		}
+	}
+	return nil
+}
+
+func (m *Mysql) createTables() {
+	_, err := m.db.Exec(`
+        CREATE TABLE IF NOT EXISTS ` + m.zonesTable + ` (
+            id INT NOT NULL AUTO_INCREMENT,
+            zone_name VARCHAR(255) NOT NULL,
+            PRIMARY KEY (id),
+            UNIQUE KEY (zone_name)
+        );
+    `)
+	if err != nil {
+		createTableCount.With(prometheus.Labels{"status": "fail", "table_name": m.zonesTable}).Inc()
+		logger.Error(err)
+	} else {
+		createTableCount.With(prometheus.Labels{"status": "success", "table_name": m.zonesTable}).Inc()
+	}
+
+	_, err = m.db.Exec(`
+        CREATE TABLE IF NOT EXISTS ` + m.recordsTable + ` (
+            id INT NOT NULL AUTO_INCREMENT,
+            zone_id INT NOT NULL,
+            hostname VARCHAR(512) NOT NULL,
+            type VARCHAR(10) NOT NULL,
```

Comment:
I wonder if the record type should be an [ENUM column](https://dev.mysql.com/doc/refman/8.0/en/enum.html) of the most common record types.

Reply:
I agree with this. I'll go back and edit this next time